gemsfx/src/main/java/com/dlsc/gemsfx/skins/EmailFieldSkin.java

```diff
@@ -3,24 +3,34 @@
 import com.dlsc.gemsfx.EmailField;
 import javafx.scene.control.SkinBase;
 import javafx.scene.control.Tooltip;
+import javafx.scene.layout.Region;
 import javafx.scene.layout.StackPane;
 import org.controlsfx.control.textfield.CustomTextField;
-import org.kordamp.ikonli.javafx.FontIcon;
 
 public class EmailFieldSkin extends SkinBase<EmailField> {
 
     public EmailFieldSkin(EmailField field) {
         super(field);
 
-        FontIcon validationIcon = new FontIcon();
+        CustomTextField customTextField = field.getEditor();
 
-        StackPane validationIconWrapper = new StackPane(validationIcon);
-        validationIconWrapper.getStyleClass().add("icon-wrapper");
-        Tooltip.install(validationIconWrapper, new Tooltip("Email address is invalid"));
-        validationIconWrapper.visibleProperty().bind(field.validProperty().not());
-        validationIconWrapper.managedProperty().bind(field.validProperty().not());
+        Region mailIcon = new Region();
+        mailIcon.getStyleClass().add("mail-icon");
+        StackPane leftIconWrapper = new StackPane(mailIcon);
+        leftIconWrapper.getStyleClass().add("mail-icon-wrapper");
+        leftIconWrapper.managedProperty().bind(leftIconWrapper.visibleProperty());
+        leftIconWrapper.visibleProperty().bind(field.showMailIconProperty());
+        leftIconWrapper.visibleProperty().addListener(it -> customTextField.requestLayout());
 
-        CustomTextField customTextField = field.getEditor();
+        Region rightIcon = new Region();
+        rightIcon.getStyleClass().add("validation-icon");
+        StackPane rightIconWrapper = new StackPane(rightIcon);
+        rightIconWrapper.getStyleClass().add("validation-icon-wrapper");
+        Tooltip.install(rightIconWrapper, new Tooltip("Email address is invalid"));
```

Comment:
No tooltip here or introduce a StringProperty on the control that lets the developer specify the text. The way it is right no does not support i18n.